diff.py

```diff
@@ -1606,6 +1638,49 @@ def _normalize_adrp_differences(self, mnemonic: str, row: str) -> str:
         return row
 
 
+class AsmProcessorI686(AsmProcessor):
+    def process_reloc(self, row: str, prev: str) -> Tuple[str, Optional[str]]:
+        arch = self.config.arch
+        repl = row.split()[-1]
+        args_string = ""
+        mnemonic, args = prev.split(maxsplit=1)
+
+        comma = args.find(",")
+        if comma != -1:
+            args_string = args[comma:]
+
+        if "R_386_NONE" in row:
+            pass
+        elif "R_386_32" in row:
+            pass
+        elif "R_386_PC32" in row:
+            pass
+        elif "R_386_16" in row:
+            pass
+        elif "R_386_PC16" in row:
+            pass
+        elif "R_386_8" in row:
+            pass
+        elif "R_386_PC8" in row:
+            pass
+        elif "R_386_GOT32" in row:
+            repl = f"%got({repl})"
+        elif "R_386_PLT32" in row:
+            repl = f"%plt({repl})"
+        elif "R_386_RELATIVE" in row:
+            repl = f"%rel({repl})"
+        elif "R_386_GOTOFF" in row:
+            repl = f"%got({repl})"
+        elif "R_386_GOTPC" in row:
+            repl = f"%got({repl})"
+        elif "R_386_32PLT" in row:
+            repl = f"%plt({repl})"
+        else:
+            assert False, f"unknown relocation type '{row}' for line '{prev}'"
+
+        return f"{mnemonic}\t{repl}{args_string}", repl
```

Comment:
the relocated part of the instruction doesn't have to be the first one: https://godbolt.org/z/z6fhMvzse (x86_64, but I think x86 is the same)